lib/executor/ramping_arrival_rate.go

```diff
@@ -316,25 +316,52 @@ func (varr RampingArrivalRate) Run(parentCtx context.Context, out chan<- stats.S
 	returnedVUs := make(chan struct{})
 	startTime, maxDurationCtx, regDurationCtx, cancel := getDurationContexts(parentCtx, duration, gracefulStop)
 
+	// Iterations chan notifies a new available iteration request.
+	// The number of active VU iterators and
+	// the rate computed should be able to maintain this chan empty most of the time.
+	// If the chan is not empty then
+	// it means the sender rate is not matching the VUs availability.
+	iterations := make(chan struct{}, 1)
+	defer close(iterations)
+
+	// exitVUIterator notifies when an iterator must stop to process.
+	// So, a message on the chan decreases the number of active iterators.
+	exitVUIterator := make(chan struct{}, maxVUs)
+
 	defer func() {
 		// Make sure all VUs aren't executing iterations anymore, for the cancel()
 		// below to deactivate them.
 		<-returnedVUs
 		cancel()
 		activeVUsWg.Wait()
+		close(exitVUIterator)
 	}()
+
 	activeVUs := make(chan lib.ActiveVU, maxVUs)
 	activeVUsCount := uint64(0)
 
 	returnVU := func(u lib.InitializedVU) {
+		exitVUIterator <- struct{}{}
```

Comment:
The VU you're returning here might not be the same VU that receives the signal further down on line 360.